### DISCUSSION TITLE
How to build download link pointing to my own view

### DISCUSSION THREAD

**Discussion by MShekow** [lona-web-org/lona#155] [Q&A] [answered]
Hi. I've built a view (defined with `http_pass_through=True` in `routes.py`) that serves files that the user wants to download. This download-view computes the file content on-demand (I am aware that for _static_ file serving, there are better alternatives, like using a proper HTTP server such as Apache).

In another view, I'm trying to have a clickable link (`A`) that will download the file from my download view.

What does not work well is to define something like `A("download", href="my-download-view/")` because that will trigger a view switch, which I don't want.

If I do something like `A("download", href="my-download-view/", events=[CLICK])` then the unwanted view-switch no longer occurs, but then a normal click simply does nothing. The user has to either force opening the link in a new tab (and has to do it themselves e.g. via shift+click, because something like  `A(... target="_blank" ...)` does not open a new tab for unknown reasons), or they have to right-click -> download the link.

How can I achieve that a simple, single click on the link triggers the download?

**Comment by fscherf** [comment, marked as answer]
@MShekow what you are searching for is ``interactive``. You can tell the javascript client to not change the behavior of links by setting ``interactive`` to ``False``

``A('foo', href='/foo.pdf', interactive=False)``

**Reply by MShekow** [reply]
Thank you, that worked.

**Comment by MShekow** [comment]
I found another small problem: once the user clicks on a download link (created with `interactive=False`) Lona no longer handles any (interactive) follow-up link-clicks anymore.

**Reply by fscherf** [reply]
Oh! That sounds like a bug. Do you have a reproducer for that?

**Reply by MShekow** [reply]
Ahh, I found that the problem is that my browser simply terminates all connections (including lona's websocket) when clicking on a download link. The problem is solved via something like `A('foo', href='/foo.pdf', interactive=False, target='_blank')`.

**Reply by fscherf** [reply]
Ah ok. Maybe that's something for the documentation

**Reply by MShekow** [reply]
Maybe you could simply add a third example to https://lona-web.org/end-user-documentation/html.html#links e.g.
```
# internal link to a non-interactive (or HTTP-pass-through) view that serves a downloadable file.
# Without "target='_blank'", the browser would try to download the file in the current browser
# tab, terminating the websocket connection, which would break the currently opened, interactive view
A('Internal download Link', href='/internal-download-link/', interactive=False, target='_blank')
```

**Reply by fscherf** [reply]
@MShekow Great! Thanks for that, i will add it to the documentation

Edit: It's rolled out to lona-web.org
